script.js

```diff
@@ -0,0 +1,70 @@
+let inputauthor = document.querySelector(".inputauthor")
+let inputtitle = document.querySelector(".inputtitle")
+let inputpage = document.querySelector(".inputpages")
+let container = document.querySelector(".container")
+let addbookbutton = document.querySelector(".addbook")
+let status1 = document.querySelector(".status1")
+let status2 = document.querySelector(".status2")
+addbookbutton.addEventListener("click",newbook)
+let defaultbook1 = new bookcreator("jk rowling","harry potter",1024)
+let bookarray = [defaultbook1]
+function bookcreator(author,title,pages){
+    this.author = author
+    this.title = title
+    this.pages = pages
+}
+bookcreator.prototype.deletethis = function(){
+    delete this.author
+    delete this.title
+    delete this.page
+}
+function clearinput(){
+    inputauthor.value = ""
+    inputtitle.value = ""
+    inputpage.value = ""
+}
+function newbook(){ 
+    if(inputauthor.value === "" || inputtitle.value === "" || inputpage.value === ""){
+        alert("fill all fields")
+    }
+    else{
+        let createbook = new bookcreator(inputauthor.value,inputtitle.value,inputpage.value)
+        bookarray.push(createbook)
+            let newdiv = document.createElement("div")
+            container.appendChild(newdiv)
+            for(let i = 0 ; i <= 4  ; i++){
+                let newspan = document.createElement("span")
+                newdiv.appendChild(newspan)
+            }
```

Comment:
How about create each span with it's dedicated content and then append it to `newdiv`? That way you don't have to come back to look them up using the many `document.querySelector(".container > :last-child > :nth-child(x)")` below.

You could even make it generic if you like with something like:
```js
function appendSpanToParentWithContent(parent, content) { // generic child span creator
   const span = document.createElement("span")
   span.textContent = content
   parent.appendChild(span)

  // return span // optional return span
} 


// then you could use it
appendSpanToParentWithContent(newDiv, author) // add author
appendSpanToParentWithContent(newDiv, title). // add title
// ... add the rest
//
// or if you really want to make things terse
[author, title].forEach((content) => appendSpanToParentWithContent(newDiv, content))

```